be/src/common/logconfig.cpp

```diff
@@ -207,6 +209,58 @@ static void failure_function() {
     std::abort();
 }
 
+// Calculate timezone offset string dynamically
+static std::string get_timezone_offset_string(const google::LogMessageTime& time) {
+    char tz_str[7] = "+0000";
+
+    // Get timezone offset from LogMessageTime
+    long offset_seconds = time.gmtoffset().count();
+
+    int offset_hours = static_cast<int>(offset_seconds / 3600);
+    int offset_mins = static_cast<int>((offset_seconds % 3600) / 60);
+    if (offset_mins < 0) offset_mins = -offset_mins;
+
+    if (offset_seconds < 0) {
+        snprintf(tz_str, sizeof(tz_str), "-%02d%02d", -offset_hours, offset_mins);
+    } else {
+        snprintf(tz_str, sizeof(tz_str), "+%02d%02d", offset_hours, offset_mins);
+    }
+
+    return std::string(tz_str);
+}
+
+// Custom prefix formatter that includes timezone information
+static void custom_prefix_formatter(std::ostream& s, const google::LogMessage& message, void*) {
+    const google::LogMessageTime& time = message.time();
+
+    // Severity level (single character)
+    s << google::GetLogSeverityName(message.severity())[0];
+
+    // Date and time
+    if (FLAGS_log_year_in_prefix) {
+        // Format: YYYYMMDD HH:MM:SS.uuuuuu +HHMM
+        s << std::setfill('0') << std::setw(4) << (1900 + time.year()) << std::setw(2) << (1 + time.month())
+          << std::setw(2) << time.day() << ' ' << std::setw(2) << time.hour() << ':' << std::setw(2) << time.min()
+          << ':' << std::setw(2) << time.sec() << '.' << std::setw(6) << time.usec();
+    } else {
+        // Format: MMDD HH:MM:SS.uuuuuu +HHMM
+        s << std::setfill('0') << std::setw(2) << (1 + time.month()) << std::setw(2) << time.day() << ' '
+          << std::setw(2) << time.hour() << ':' << std::setw(2) << time.min() << ':' << std::setw(2) << time.sec()
+          << '.' << std::setw(6) << time.usec();
+    }
+
+    // Timezone offset (calculated dynamically)
+    s << ' ' << get_timezone_offset_string(time);
```

Comment:
this calculation will do for every line, wondering if there is perf bench due to this logformat change.

Also this logline format change should be marked as "behavior change".